scripts/packages/VSCodeServer/src/VSCodeServer.jl

```diff
@@ -193,126 +201,146 @@ function serve(args...; is_dev = false, crashreporting_pipename::Union{AbstractS
     conn_endpoint[] = JSONRPC.JSONRPCEndpoint(conn, conn)
     run(conn_endpoint[])
 
-    if is_dev
-        @async while true
-            try
-                Base.invokelatest(handle_message)
-            catch err
-                Base.display_error(err, catch_backtrace())
-            end
-        end
-    else
-        @async try
-            while true
-                Base.invokelatest(handle_message; crashreporting_pipename = crashreporting_pipename)
+    @async try
+        msg_dispatcher = JSONRPC.MsgDispatcher()
+
+        msg_dispatcher[repl_runcode_request_type] = repl_runcode_request
+        msg_dispatcher[repl_getvariables_request_type] = repl_getvariables_request
+        msg_dispatcher[repl_getlazy_request_type] = repl_getlazy_request
+        msg_dispatcher[repl_showingrid_notification_type] = repl_showingrid_notification
+        msg_dispatcher[repl_loadedModules_request_type] = repl_loadedModules_request
+        msg_dispatcher[repl_isModuleLoaded_request_type] = repl_isModuleLoaded_request
+        msg_dispatcher[repl_startdebugger_notification_type] = (conn, params)->repl_startdebugger_request(conn, params, crashreporting_pipename)
```

Comment:
I'm generally weary of global stuff, and have a hunch that it might be easier to write tests for all of this if we have less global state. But I'm not sure, to be honest... I don't think the startup cost would make much of difference: we would save adding like a handful of items to a dict, that can't take much time, right?

Reply:
Yeah, let's keep these here.

Reply:
yeah, now I think there is no strong motivation to make them global. sorry for the noise.